src/amo/api/shelves.js

```diff
@@ -0,0 +1,17 @@
+/* @flow */
+import { callApi } from 'core/api';
+import type { ApiState } from 'core/reducers/api';
+import type { ExternalSponsoredShelfType } from 'amo/reducers/shelves';
+
+export type GetSponsoredShelfParams = {| api: ApiState |};
+
+export const getSponsoredShelf = ({
+  api,
+}: GetSponsoredShelfParams): Promise<ExternalSponsoredShelfType> => {
+  return callApi({
+    apiState: api,
+    auth: true,
```

Comment:
Is this really needed?

Reply:
I don't think we do, but I modeled this after existing API code for `recommendations`. I guess the problem exists there too. I can remove it from here.